components/infra-proxy-service/server/v1/environments.go

```diff
@@ -0,0 +1,87 @@
+package v1
+
+import (
+	"context"
+	"encoding/json"
+	"sort"
+
+	chef "github.com/chef/go-chef"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/chef/automate/api/interservice/infra_proxy/request"
+	"github.com/chef/automate/api/interservice/infra_proxy/response"
+)
+
+// GetEnvironments get environments list
+func (s *Server) GetEnvironments(ctx context.Context, req *request.Environments) (*response.Environments, error) {
+
+	client, err := s.createClient(ctx, req.OrgId)
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "invalid org id: %s", err.Error())
+	}
+
+	environments, err := client.Environments.List()
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, err.Error())
+	}
+
+	return &response.Environments{
+		Environments: fromAPIToListEnvironments(*environments),
+	}, nil
+}
+
+// GetEnvironment get data bag
+func (s *Server) GetEnvironment(ctx context.Context, req *request.Environment) (*response.Environment, error) {
+	client, err := s.createClient(ctx, req.OrgId)
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "invalid org id: %s", err.Error())
+	}
+
+	en, err := client.Environments.Get(req.Name)
+	if err != nil {
+		return nil, status.Error(codes.InvalidArgument, err.Error())
+	}
+
+	defaultAttributes, _ := json.Marshal(en.DefaultAttributes)
```

Comment:
Specific data type to hold `DefaultAttributes` data?

Reply:
Took the reference from `NodeAttribute` for dynamic attributes as:

https://github.com/chef/automate/blob/440f656fc4d396b40e1e70386e56d42e93f92e16/api/external/cfgmgmt/response/nodes.proto#L65-L72